sig-robotics/proposal/Definition of cloud robot CRD and Operator/Definition of cloud robot CRD and Operator.md

```diff
@@ -0,0 +1,157 @@
+
+## Design Details
+
+### Architecture
+
+For this project, the architecture adopted basically follows the overall architecture of KubeEdge. The structure diagram of the project is shown in the figure below
+
+<img src="./images/architecture.png" alt="architecture" style="zoom:90%;" />
+
+The current predominant approach for managing cloud robots is to treat the robot as a unified entity, considering it as a Device and accessing the KubeEdge platform for management and scheduling purposes.
+
+However, given that a robot is inherently composed of diverse sensors and controllers, it is more appropriate to view it as a combination of heterogeneous components rather than an indivisible arrangement. Consequently, in this solution, a robot is treated as an edge node, with its sensors and controllers serving as the device access platform for that node.
+
+## CRD design
+
+The purpose of this proposal is to propose a set of universal CRD and Operator for cloud robot. Since most industrial robot companies in China have joined the Mobile Robot Industry Alliance, the data interface specifications for robots and their scheduling systems proposed by them are widely applicable. This proposal refers to this specification to design the robot CRD.
+
+To achieve our goal, we need three kinds of CRDs: **robot CRD**, **task CRD** and **robotSync CRD**. The robot CRD and task CRD describe the robot status and task information respectively, and the robotSync CRD is used to record the number of robot CR and their configuration informations. When the key information of the robot changes, the status information must be reported immediately, otherwise the information is reported in the form of a heartbeat frame at a period set by the user. Task resources are created by users and delivered by the Controller.
+
+- The fields contained in the **robot** resource can be described as the table below:
+
+|        Fields       |   Type   |              Description               |
+| :-----------------: | :------: | :------------------------------------: |
+|       robotID       |   uint   |            Unique ID of a robot        |
+| registration_status | boolean  |  Indicates whether robot is registrated  |
+|      position       |  struct  | Indicates the location of the robot, including coordinates, the last point passed, etc. |
+|   running_status    |  struct  | Represents information related to robot motion, including linear velocity and angular velocity |
+|     task_info     |  struct  | Including the identification of orders and tasks, a task can be represented by a series of points and segments |
+|    battery_status    |  struct  | Including information such as remaining battery power, charging status, etc. |
+|   abnormal_events   | struct[] |         Record the exception information of the node         |
+|    robot_status     | boolean  |  indicates whether robot is ready for scheduling  |
+
```

Comment:
**Suggested Addition:** Resource Definition - CPU, GPU, Memory, etc.

To better support resource management and scheduling for cloud robots, it is recommended to add resource-related definitions, including CPU, GPU, and memory, to the CRD. These resource definitions will assist the system in efficiently allocating and managing computing resources for cloud robots, thereby enhancing system performance and reliability.